### DISCUSSION TITLE
Is there a fast way to send function defintions?

### DISCUSSION THREAD

**Discussion by SimonCW** [Vigemus/iron.nvim#264] [Q&A] [answered]
Hey, 
is there a fast way to send function definitions? I cannot figure out how you do it in the gif from the Readme https://asciinema.org/a/495376.

For a function
```python
def fn1(a, b):
    ```This is doing something````
    return 3*a+b

```
When finished typing the return statement, I'd like to send the function to ipython. 
Currently, I type `<space>sc` (to send a motion) and then `3k` to send the 3 lines up. Alternatively, I could do `<space>sc{` 

Is there a more ergonomic way to do this?

**Comment by jolars** [comment, marked as answer]
Yes, via treesitter textobjects: https://github.com/nvim-treesitter/nvim-treesitter-textobjects

So if you have mapped, for instance, `gs` as send motion in iron and `af` for `@function.outer` for treesitter, you could just do `gsaf`.

**Reply by kraxli** [reply]
@jolars how would this work for ` R` ?

It seems that the suggested approach only sends, e.g.
```R
function(x) {
  return(x^2)
}
```

without the function name instead of:

```R
square <- function(x) {
  return(x^2)
}
```

**Reply by jolars** [reply]
Yes, that's a problem with how the function text object is defined in the treesitter parser for R. I agree it's annoying. I am guessing you might be able to redefine or use some plugin to handle this, to include the nodes for the assignment operator and object the function is assigned to, but I haven't experimented with this.
